docs/TuxReact.md

```diff
@@ -0,0 +1,449 @@
+
+***
+
+##### <a id="React-Components-props-children"></a>3.3.1) props.children [#](#React-Components-props-children)
+When you pass in children to a `React` component you are actually just passing in `props` to the component through an altered syntax. When you pass in children to a component, that component will have access to them through the opaque data structure `this.props.children`.
+
+**NOTE** that this means that passing in children to a component does not mean that those children will be rendered as is. It's up to the component to decide what to do with its children. For the base `React` components children will be rendered as is however. You can read more about children and ownership in `React` [here](http://facebook.github.io/react/docs/multiple-components.html).
+
+The `this.props.children` data structure is opaque so you should use `React` specialized methods for accessing `children`. Those methods are described in full [here](http://facebook.github.io/react/docs/top-level-api.html#react.children)
+
+```
+    var Parent = React.createClass({
+      render: function () {
+        <div>
+          { this.props.children }
+        </div>
+      }
+    });
+```
+
+***
+
+##### <a id="React-Components-PropTypes"></a>3.3.2) PropTypes [#](#React-Components-PropTypes)
+Passing `props` down into components is a somewhat passive framework design. To combat this, components can define the `propTypes` they need or expect:
+
+```
+   Var MyComponent = React.createClass({
+     propTypes: {
+       optionalObjectProp: React.PropTypes.object,
+       requiredFunctionProp: React.PropTypes.func.isRequired
+     }
+   });
+```
+
+There is a huge laundry list `propTypes` specifications you can provide. Review them [here](http://facebook.github.io/react/docs/reusable-components.html#prop-validation).
+
+***
+
+#### <a id="React-Components-Lifecycle-Methods"></a>3.4) Lifecycle Methods [#](#React-Components-PropTypes)
+The main way to hook into `React` components is throught lifecycle methods which are triggered at different significant events for the component. Lifecycle events are triggered in a batched depth first manner.This is easiest to show with an example:
+
+```
+    var MyLifeCycleComponent = React.createClass({
+      componentWillMount: function () {
+        //I am called once and first, the first time the component is going to render
+        //if setState is invoked in me then it will not trigger a rerender but the upcoming render will see the new state
+      },
+
+      componentDidMount: function () {
+        //I am called once after the initial render, you can access the DOM representation of the component for the first time with this.getDOMNode() in me
+      },
+
+      componentWillReceiveProps: function (newProps) {
+        //I am called every time the component is receiving properties after the initial render
+        //I have access to the upcoming props through the newProps input.
```

Comment:
Recommend removing period here if you aren't using them everywhere else, either that or apply a periods at the end of all the comments.